Orleans/Test/TinkerHostTests/TinkerHostLifecycleTests.cs

```diff
@@ -83,7 +95,7 @@ public class Thingy : ILifecycleObserver
             public Thingy(ILifecycleObservable lifecycle, ILoggerFactory loggerFactory)
             {
                 logger = loggerFactory.CreateLogger<Thingy>();
-                lifecycle.Subscribe(this);
+                lifecycle.Subscribe(this); //<-- passing this before finishing constructing is dangerous (although practical)
```

Comment:
yeah, I was being lazy.  There are extension functions to connect callbacks.  I'll put out a PR to illustrate.